README.md

```diff
@@ -919,6 +919,42 @@ parsers, it may be more convenient to use the `xml_mode` context manager:
     # ... We have now returned to using the default XML parser.
 ```
 
+Application Cleanup
+-------------------
+
+A user-defined function may be registered to be invoked before, during, or after the time
+when the Python iRODS Client is engaged in object teardown (also called "cleanup") operations.
+This is analogous to the atexit facility in C or in Python itself, except that we have the extra
+resolution to specify that a function or callable object be run either just before, or just after, the
+aforementioned object teardown stage:
+
+```
+    from irods import at_client_exit
+    at_client_exit.register_for_execution_after_prc_cleanup(lambda : print("PRC cleanup has completed."))
+    at_client_exit.register_for_execution_before_prc_cleanup(lambda : print("PRC cleanup is about to start."))
+```
+Alternatively, we can use a lower-level interface to access the "during" aspect of things and run a function closer in
+time to the actual Python client cleanup processing.
+```
+    at_client_exit._register(LibraryCleanupStage.DURING, lambda : print("This is printed during the PRC cleanup stage."))
+```
+A function normally cannot be registered multiple times to run in the same stage, but we may be overcome this limitation
+(and, optionally, arguments set for the invocation) by wrapping the same function into two different callable objects:
+```
+    for i in (1,2):
+        at_client_exit.register_for_execution_after_prc_cleanup(
+            at_client_exit.unique_function_invocation(lambda n: print(f"Called just before exit - iter {n}"), tup_args = (i,))
+            )
+```
```

Comment:
Some comments on this block:

1. Please use "```python" on L943 for syntax highlighting
2. The "just before exit" language registered as "after" PRC cleanup confused me a little bit. Maybe it should just say "just after PRC cleanup" instead?
3. Just to make sure I understand, if this was not wrapped in `unique_function_invocation`, it would not register multiple `lambda` functions? In other words, in order for this to work, you *must* wrap each `lambda` in the "after" stage with the `unique_function_invocation`?

```suggestion
```python
    for i in (1,2):
        at_client_exit.register_for_execution_after_prc_cleanup(
            at_client_exit.unique_function_invocation(lambda n: print(f"Called just after PRC cleanup - iter {n}"), tup_args = (i,))
            )
```

Reply:
Yes, right.  There are plenty of other places in the README we might want to do that as well.  Perhaps another issue?

Reply:
> 3. Just to make sure I understand, if this was not wrapped in `unique_function_invocation`, it would not register multiple `lambda` functions? In other words, in order for this to work, you _must_ wrap each `lambda` in the "after" stage with the `unique_function_invocation`?

Correct.  Mainly because the primary imagined use of this is to register a single-invocation system cleanup function from within another library that depends on , but clearly rests on top of, PRC abstractions.  So, I made repetitive registrations explicit using the `unique_function_invocation` (only a part of that class's utility, I'll remind everyone.)  

Which I think is all-around less confusing.  I.e., why are you registering a function for execution within XYZ stage, if it's already registered.

Reply:
resolving this.